### PR TITLE
Changing the signal handling to named pipe in the group test for better portablity (2nd improvement)

### PR DESCRIPTION
**Pull request by raylu-hdf**
Rearranged the test in the following way:

1. Reader waits for a number of ticks (3 by default) before verifying the creation of groups and attributes;
2. Error handling is consistent with traditional setup of the test suite.

### CODE REVIEW

File: test/vfd_swmr_group_writer.c [hunk 1]
```diff
@@ -282,64 +367,257 @@ main(int argc, char **argv)
...
+    H5E_BEGIN_TRY {
+        H5Pclose(fapl);
+        H5Pclose(fcpl);
+        H5Fclose(s.file);
+    } H5E_END_TRY;
+
+    if (fd_writer_to_reader >= 0)
+        HDclose(fd_writer_to_reader);
+
+    if (fd_writer_to_reader >= 0)
```
Comment (kyang2014) [inline test/vfd_swmr_group_writer.c]: This is a typo. It should be  

`if(fd_reader_to_writer >= 0)`

File: test/vfd_swmr_group_writer.c [hunk 2]
```diff
@@ -282,64 +367,257 @@ main(int argc, char **argv)
...
+                    notify = -1;
+                    HDwrite(fd_writer_to_reader, &notify, sizeof(int));
+                    goto error;
+                }
+            } else {
+                /* At communication interval, notifies the reader and waits for its response */
+                if (step % s.csteps == 0) {
+                    /* Bump up the value of notify to notice the reader to start to read */
+                    notify++;
+                    if (HDwrite(fd_writer_to_reader, &notify, sizeof(int)) < 0) {
```
Comment (vchoi-hdfgroup) [inline test/vfd_swmr_group_writer.c:464]: There are sections of code in "named pipe handling" that are common in both writer and reader.
Is it possible to put those common coding in routine(s) for clarification purpose?

File: test/testvfdswmr.sh.in [hunk 3]
```diff
@@ -610,25 +610,24 @@ if [ ${do_zoo:-no} = yes ]; then
...
 if [ ${do_groups:-no} = yes ]; then
 	echo launch vfd_swmr_group_writer
 	catch_out_err_and_rc vfd_swmr_group_writer \
-	    ../vfd_swmr_group_writer -q -u 10 -n 10000 &
+	    ../vfd_swmr_group_writer -q -c 100 -n 1000 &
 	pid_writer=$!
 
 	catch_out_err_and_rc vfd_swmr_group_reader \
-	    ../vfd_swmr_group_reader -q -u 10 -n 10000 -W &
+	    ../vfd_swmr_group_reader -q -c 100 -n 1000 -u 5 &
```
Comment (vchoi-hdfgroup) [inline test/testvfdswmr.sh.in]: Per our discussion, due to the slowness in running 10000 groups based on the current changes, we may need to put the group test also in HDF5TestExpress  setting in testvfdswmr.sh.in.

File: test/vfd_swmr_group_writer.c (hunk 2 above)
Comment (raylu-hdf) [reply test/vfd_swmr_group_writer.c:464]: I looked at code again and didn't feel refactoring would help much.

File: test/testvfdswmr.sh.in (hunk 3 above)
Comment (raylu-hdf) [reply test/testvfdswmr.sh.in]: Yes, I agree and changed the script.

**Review by kyang2014 — APPROVED** [review]

**Review by vchoi-hdfgroup — APPROVED** [review]

File: test/vfd_swmr_group_writer.c (hunk 1 above)
Comment (raylu-hdf) [reply test/vfd_swmr_group_writer.c]: I fixed it.  Thanks.

**Pull request merged**